File: app/src/main/java/org/dhis2/usescases/notes/NotesAdapter.java
package org.dhis2.usescases.notes;

import java.util.ArrayList;
import java.util.List;

import org.dhis2.R;
import org.dhis2.data.tuples.Pair;
import org.dhis2.databinding.ItemNotesBinding;
import org.hisp.dhis.android.core.note.Note;
import org.jetbrains.annotations.NotNull;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import io.reactivex.processors.FlowableProcessor;
import io.reactivex.processors.PublishProcessor;

/**
 * QUADRAM. Created by Administrador on 18/12/2017.
 */

public class NotesAdapter extends RecyclerView.Adapter<NotesViewholder> {

    private List<Note> notes;
    private final FlowableProcessor<Pair<String, Boolean>> processor;

    public NotesAdapter() {
        this.processor = PublishProcessor.create();
        this.notes = new ArrayList<>();
    }

    @NotNull
    @Override
    public NotesViewholder onCreateViewHolder(@NotNull ViewGroup parent, int viewType) {
        ItemNotesBinding itemNotesBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()), R.layout.item_notes, parent, false);
        return new NotesViewholder(itemNotesBinding);
    }

    @Override
    public void onBindViewHolder(@NotNull NotesViewholder holder, int position) {
        holder.bind(notes.get(position));
    }

    @Override
    public int getItemCount() {
        return notes.size();
    }

    public void setItems(List<Note> notes) {
        this.notes = notes;
        notifyDataSetChanged();
    }

    @NonNull
    public FlowableProcessor<Pair<String, Boolean>> asFlowable() {
        return processor;
    }
}

File: app/src/main/java/org/dhis2/usescases/notes/NotesAdapter.kt
/*
 * Copyright (c) 2004-2019, University of Oslo
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 * Redistributions of source code must retain the above copyright notice, this
 * list of conditions and the following disclaimer.
 *
 * Redistributions in binary form must reproduce the above copyright notice,
 * this list of conditions and the following disclaimer in the documentation
 * and/or other materials provided with the distribution.
 * Neither the name of the HISP project nor the names of its contributors may
 * be used to endorse or promote products derived from this software without
 * specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
 * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
 * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
 * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
 * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
 * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
package org.dhis2.usescases.notes

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import io.reactivex.processors.FlowableProcessor
import io.reactivex.processors.PublishProcessor
import java.util.ArrayList
import org.dhis2.R
import org.dhis2.data.tuples.Pair
import org.dhis2.databinding.ItemNoteBinding
import org.hisp.dhis.android.core.note.Note

class NotesAdapter : RecyclerView.Adapter<NotesViewHolder>() {

    private var notes: List<Note> = ArrayList()
    private val processor: FlowableProcessor<Pair<String, Boolean>>

    init {
        this.processor = PublishProcessor.create()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NotesViewHolder {
        val itemNotesBinding = DataBindingUtil.inflate<ItemNoteBinding>(
            LayoutInflater.from(parent.context),
            R.layout.item_note,
            parent,
            false
        )
        return NotesViewHolder(itemNotesBinding)
    }

    override fun onBindViewHolder(holder: NotesViewHolder, position: Int) {
        holder.bind(notes[position])
    }

    override fun getItemCount(): Int = notes.size

    fun setItems(notes: List<Note>) {
        this.notes = notes
        notifyDataSetChanged()
    }

    fun asFlowable(): FlowableProcessor<Pair<String, Boolean>> = processor
}
